src/app/components/board/index.tsx

```diff
@@ -33,4 +39,101 @@ const Board = (props: BoardProps) => {
   }
 };
 
+const Board = (props: BoardProps) => {
+  const { questions } = props;
+  const { course } = useOfficeHour();
+  const topics = [SELECT_ALL, ...Object.keys(course.tags).sort()];
+
+  const [selectedTopics, setSelectedTopics] = React.useState([SELECT_ALL]);
+
+  const getQuestionsByTopic = (topics: string[]) => {
+    if (topics.length === 0) {
+      return [];
+    } else if (topics.includes(SELECT_ALL)) {
+      return questions;
+    } else {
+      return questions.filter((question) =>
+        question.tags.find((tag) => topics.includes(tag))
+      );
+    }
+  };
+
+  const addOrRemoveTopic = (
+    _e: React.MouseEvent<HTMLElement>,
+    newTopics: string[]
+  ) => {
+    setSelectedTopics((currentSelectedTopics) => {
+      if (currentSelectedTopics.includes(SELECT_ALL)) {
+        return newTopics.filter((topic) => topic !== SELECT_ALL);
+      } else if (newTopics.includes(SELECT_ALL)) {
+        return [SELECT_ALL];
+      } else {
+        return newTopics;
+      }
+    });
+  };
+  return (
+    <>
+      <ToggleButtonGroup value={selectedTopics} onChange={addOrRemoveTopic}>
```

Comment:
Currently for some of the phones on my end (e.g. iPhone 12 Pro) if you click on all the options for CS160 it forces the entire screen to be scrollable in both the x and y directions. Could we make it so that the tag options are scrollable horizontally instead?
<img width="1680" alt="Screenshot 2024-08-18 at 4 57 51 PM" src="https://github.com/user-attachments/assets/a30b7c24-c197-47f7-93c8-68858cc32ccf">

Reply:
Are you zooming in on the screen?

<img width="460" alt="image" src="https://github.com/user-attachments/assets/4460eae8-bcad-4895-b31b-ba8c2906a42d">

I do think though it would be nice to add a little right padding on the header. What do you think?

Reply:
Yeah right padding sounds good 👍

Initially I'm not zoomed in, but if I click both Conceptual and HW Help it forces the whole page to zoom in. Even if you zoom out the page seems to still be off-centered (there's more space on the right compared to the left):

<img width="370" alt="Screen Shot 2024-08-25 at 2 52 45 PM" src="https://github.com/user-attachments/assets/b613f568-a72a-4b7b-a0e4-b9209546f53a">
<img width="383" alt="Screen Shot 2024-08-25 at 2 52 58 PM" src="https://github.com/user-attachments/assets/cd4411b7-b5ee-4d21-82b8-84046df48508">

Reply:
Looks great now 🙌